android/fakedapp/src/androidTest/java/com/solana/mobilewalletadapter/fakedapp/MainActivityTest.kt

```diff
@@ -21,4 +25,186 @@ class MainActivityTest {
+        onView(withId(R.id.btn_sign_and_send_txn_x1)).perform(click())
+
+        handleWalletDisambiguationIfNecessary(uiDevice)
+
+        uiDevice.findObject(By.text("AUTHORIZE")).click()
+
+        uiDevice.waitForWindowUpdate(null, 500)
+
+        // send transaction to cluster is flaky and relies on a successful airdrop, which is
+        // difficult with the public devnet/testnet RPCs. Will revisit this with a local validator!
+//        uiDevice.findObject(By.textStartsWith("SEND TRANSACTION")).click()
+        uiDevice.findObject(By.textStartsWith("SIMULATE SUBMITTED")).click()
+
+        // then
+        onView(withText(R.string.msg_request_succeeded)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun authorizeAndSignTx1_V1_isSuccessful() {
+        // given
+        val uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+        // when
+        onView(withId(R.id.spinner_txn_ver)).perform(click())
+        onView(withText(R.string.string_txn_version_v0)).perform(click())
+        onView(withId(R.id.btn_authorize)).perform(click())
+
+        handleWalletDisambiguationIfNecessary(uiDevice)
+
+        uiDevice.findObject(By.text("AUTHORIZE")).click()
+
+        // back in dApp, click sign
+        onView(withId(R.id.btn_sign_txn_x1)).perform(click())
+
+        handleWalletDisambiguationIfNecessary(uiDevice)
+
+        uiDevice.findObject(By.text("AUTHORIZE")).click()
+
+        // then
+        onView(withText(R.string.msg_request_succeeded)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun authorizeAndSignAndSendTx1_V1_isSuccessful() {
+        // given
+        val uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+        // when
+        onView(withId(R.id.spinner_txn_ver)).perform(click())
+        onView(withText(R.string.string_txn_version_v0)).perform(click())
+        onView(withId(R.id.btn_authorize)).perform(click())
+
+        handleWalletDisambiguationIfNecessary(uiDevice)
+
+        uiDevice.findObject(By.text("AUTHORIZE")).click()
+
+        // back in dApp, click sign
+//        onView(withId(R.id.btn_request_airdrop)).perform(click())
+        onView(withId(R.id.btn_sign_and_send_txn_x1)).perform(click())
+
+        handleWalletDisambiguationIfNecessary(uiDevice)
+
+        uiDevice.findObject(By.text("AUTHORIZE")).click()
+
+        // send transaction to cluster is flaky and relies on a successful airdrop, which is
+        // difficult with the public devnet/testnet RPCs. Will revisit this with a local validator!
+//        uiDevice.findObject(By.textStartsWith("SEND TRANSACTION")).click()
+        uiDevice.findObject(By.textStartsWith("SIMULATE SUBMITTED")).click()
+
+        // then
+        onView(withText(R.string.msg_request_succeeded)).check(matches(isDisplayed()))
+    }
+
+    private fun handleWalletDisambiguationIfNecessary(uiDevice: UiDevice) {
+        uiDevice.findObjects(By.textContains("Fake Wallet")).forEach {
```

Comment:
these find by text calls will obviously break if the phone used to test is set to a different language. I think this can be improved by using different matching logic in the find methods.